src/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -2212,7 +2285,11 @@ public static bool TrueForAll<T>(T[] array, Predicate<T> match)
 
         public unsafe Object GetValue(int index)
         {
+#if REAL_MULTIDIM_ARRAYS
+            if (!this.EETypePtr.IsSzArray)
```

Comment:
Made it call the `IsSzArray` on `System.Array`, but now it doesn't feel safe to address the first comment ("This impl should work fine everywhere - this ifdef should not be needed."). I don't know what BaseSize for the fake MDArrays looks like and whether it's safe to make any assumptions about it.